src/mds/Server.cc

```diff
@@ -1080,21 +1080,21 @@ void Server::recover_filelocks(CInode *in, bufferlist locks, int64_t client)
  * to trim some caps, and consequently unpin some inodes in the MDCache so
  * that it can trim too.
  */
-void Server::recall_client_state(float ratio)
+void Server::recall_client_state(void)
 {
-  int max_caps_per_client = (int)(g_conf->mds_cache_size * .8);
-  int min_caps_per_client = 100;
+  /* try to recall at least 80% of all caps */
+  uint64_t max_caps_per_client = (Capability::count() * .8);
+  uint64_t min_caps_per_client = 100;
+  /* unless this ratio is smaller: */
+  double ratio = (mdcache->cache_size() / mdcache->cache_limit_memory())*0.9;
```

Comment:
If there is only one client, which has 200 caps. the function tries to trim its caps to 160. does this make sense?

Reply:
I see, the function is called when cache overfull